src/main/java/org/springframework/data/couchbase/core/ReactiveFindByAnalyticsOperationSupport.java

```diff
@@ -92,10 +94,22 @@ public ReactiveFindByAnalyticsOperationSupport(final ReactiveCouchbaseTemplate t
 								return throwable;
 							}
 						}).flatMapMany(ReactiveAnalyticsResult::rowsAsObject).map(row -> {
-							String id = row.getString("__id");
-							long cas = row.getLong("__cas");
-							row.removeKey("__id");
-							row.removeKey("__cas");
+							String id = "";
+							long cas = 0;
+							if (row.getString(TemplateUtils.SELECT_ID) == null) {
+								throw new CouchbaseException("query did not project " + TemplateUtils.SELECT_ID
```

Comment:
nit: should probably say in both exceptions "analytics query did not project".. to distinguish from a n1ql query in the logs if seen